### DISCUSSION TITLE
Is it possible to map claims from External System?

### DISCUSSION THREAD

**Discussion by dinbtechit** [zitadel/zitadel#7691] [Q&A]
How can I achieve the below flow in Zitadel?

![zitadel-mapping-claims-with-external-system](https://github.com/zitadel/zitadel/assets/17984781/c21374a5-dcdd-49f3-8867-ea41ee598446)

**Comment by jeffmccune** [comment]
ZITADEL supports custom claims, take a look at the actions [examples](https://github.com/zitadel/actions/blob/main/examples/), specifically [make_api_call.js](https://github.com/zitadel/actions/blob/main/examples/make_api_call.js)

**Reply by dinbtechit** [reply]
Thank you for sharing the examples. However, in my particular scenario, the user needs to be redirected to another website to obtain an additional custom claim. After that, they should return to Zitadel to create the auth tokens. Unfortunately, I couldn't find any example for this specific use case. 

I'm curious if Zitadel has anything in place that could make this type of integration more seamless.
